teuthology/packaging.py

```diff
@@ -473,6 +473,10 @@ def _init_from_config(self):
         self.flavor = self.job_config.get("flavor")
         self.codename = self.job_config.get("codename")
         self.os_version = self._get_version()
+        # if os_version is given, prefer version/codename derived from it
+        if self.os_version:
+            self.os_version, self.codename = \
```

Comment:
yeah....it's really "if someone supplied something they claim is an os_version, regularize it into a canonical version/codename tuple" (because os_version could be either, and I'm trying to adapt)